File: Common/StudentCode.cs
using System.Runtime.Serialization;

namespace DevSpace.Common {
	[DataContract]
	public class StudentCode : IStudentCode {
		private StudentCode() { }

		[DataMember]public int Id { get; private set; }
		[DataMember]public string Email { get; private set; }
		[DataMember]public string Code { get; private set; }
	}
}

File: FSharp.Common/StudentCode.fs
namespace DevSpace.FSharp.Common

open System.Runtime.Serialization

[<DataContract>]
type StudentCode =
  {
    [<field: DataMember>]Id:int;
    [<field: DataMember>]Email:string;
    [<field: DataMember>]Code:string
  }

  member this.WithId( value ) =
    { this with Id = value }

  member this.WithEmail( value ) =
    { this with Email = value }

  member this.WithCode( value ) =
    { this with Code = value }
